sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -218,13 +218,7 @@ static SpannerOptions buildSpannerOptions(SpannerConfig spannerConfig) {
     if (serviceFactory != null) {
       builder.setServiceFactory(serviceFactory);
     }
-    ValueProvider<String> host = spannerConfig.getHost();
-    if (host != null) {
-      String hostValue = host.get();
-      if (hostValue != null && !hostValue.trim().isEmpty()) {
-        builder.setHost(host.get());
-      }
-    }
+    builder.setHost(spannerConfig.getHostValue());
```

Comment:
If we don't set this at all (current behavior), what happens? Noting that this is a silent behavior change for users who passed in null here.

Reply:
If SpannerIO does not set the host in `SpannerOptions`, then the default host set in the Spanner client library is used, which is `https://spanner.googleapis.com`. 

The endpoint that should be used by default by Beam is `http://batch-spanner.googlapis.com`. I believe that this labels the traffic as batch traffic. 

If the `host` in SpannerIO is not set, then the default endpoint used by Beam is `http://batch-spanner.googlapis.com`. It was effectively a bug in Beam that setting a the SpannerIO `host` to null would use the default host from the client library, instead of the SpannerIO default host. 

Yes. this is effectively a change in behavior.

Reply:
Cool - just confirming that's intentional/making sure I understand the change. Thanks!